test/python/transpiler/test_basis_translator.py

```diff
@@ -0,0 +1,554 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+
+"""Test the BasisTranslator pass"""
+
+
+from numpy import pi
+
+from qiskit import QuantumRegister, ClassicalRegister, QuantumCircuit
+from qiskit.test import QiskitTestCase
+from qiskit.circuit import Gate, Parameter, EquivalenceLibrary
+from qiskit.converters import circuit_to_dag, circuit_to_instruction
+from qiskit.exceptions import QiskitError
+from qiskit.transpiler.exceptions import TranspilerError
+from qiskit.transpiler.passes.basis import BasisTranslator, UnrollCustomDefinitions
+
+
+from qiskit.extensions.standard.equivalence_library import StandardEquivalenceLibrary as std_eqlib
+
+
+class OneQubitZeroParamGate(Gate):
+    """Mock one qubit zero param gate."""
+    def __init__(self):
+        super().__init__('1q0p', 1, [])
+
+
+class OneQubitOneParamGate(Gate):
+    """Mock one qubit one  param gate."""
+    def __init__(self, theta):
+        super().__init__('1q1p', 1, [theta])
+
+
+class OneQubitTwoParamGate(Gate):
+    """Mock one qubit two param gate."""
+    def __init__(self, phi, lam):
+        super().__init__('1q2p', 1, [phi, lam])
+
+
+class TwoQubitZeroParamGate(Gate):
+    """Mock one qubit zero param gate."""
+    def __init__(self):
+        super().__init__('2q0p', 2, [])
+
+
+class VariadicZeroParamGate(Gate):
+    """Mock variadic zero param gate."""
+    def __init__(self, num_qubits):
+        super().__init__('vq0p', num_qubits, [])
+
+
+class TestBasisTranslator(QiskitTestCase):
```

Comment:
can you add a test where the library has two entries for a gate resulting in two paths to a target basis? Maybe something like a diamond:

     src
/              \
A,B         C,D
 \            /
     target

Reply:
Added in a11fb98 .